### PR TITLE
Properly display priority sorting in all paths and fall back to default sorting

### LINKED ISSUE #4504: Sort widget missing if sort param present in request
at least if you sort on priority.

there may also be an issue with a sort option missing

### PR DESCRIPTION
**Pull request by unho**
Fixes #4504

### CODE REVIEW

File: pootle/apps/pootle_translationproject/views.py [hunk 1]
```diff
@@ -380,8 +380,7 @@ def display_vfolder_priority(self):
         vfolder = self.extracted_path[0]
         if vfolder:
             return False
-        return VirtualFolderTreeItem.objects.filter(
-            pootle_path__startswith=self.object.pootle_path).exists()
+        return self.object.has_vfolders
```
Comment (phlax) [inline pootle/apps/pootle_translationproject/views.py]: this is what we removed - i really want to find an efficient way of doing this

**Comment by ta2-1** [comment]
lgtm

File: pootle/static/js/editor/app.js [hunk 2]
```diff
@@ -376,7 +378,8 @@ PTL.editor = {
           this.category = params.category;
         }
         if ('sort' in params) {
-          this.sortBy = params.sort;
+          const { sort } = params;
+          this.sortBy = ALLOWED_SORTS.indexOf(sort) !== -1 ? sort : 'default';
```
Comment (unho) [inline pootle/static/js/editor/app.js]: Just noticed that this includes `priority` sorting which might not necessarily be present in the dropdown.

Comment (unho) [inline pootle/static/js/editor/app.js]: This is now fixed.

**Comment by julen** [comment]
Thank you @unho, this is GTM from my side.

**Comment by unho** [comment]
@julen Thank you very much for your help!

File: tests/virtualfolder_helpers.py [hunk 3]
```diff
@@ -60,3 +63,6 @@ def test_extract_vfolder_from_path():
     # locations, but the VirtualFolderTreeItem pootle_path is unique, thus only
     # one exists.
     assert 1 == VirtualFolderTreeItem.objects.filter(pootle_path=path).count()
+
+    vfolder0.delete()
```
Comment (phlax) [inline tests/virtualfolder_helpers.py]: these 2 lines are unnecessary

**Comment by phlax** [comment]
@unho - minor point raised above - really nice work! lgtm

File: tests/virtualfolder_helpers.py (hunk 3 above)
Comment (unho) [inline tests/virtualfolder_helpers.py]: They were necessary in my local install since otherwise test crashed. Will keep them.

Comment (phlax) [inline tests/virtualfolder_helpers.py]: @unho these 2 lines server **absolutely** no purpose please remove before landing

**Pull request merged**